src/test/java/com/google/sps/CommentAnalysisTest.java

```diff
@@ -47,47 +49,57 @@
 /** This is a JUnit test for sentiment mockedAnalysis */
 @RunWith(JUnit4.class)
 public class CommentAnalysisTest {
-  private static final double LOWER_END_VAL = -1.0;
-  private static final double UPPER_END_VAL = 1.0;
-  private static final BigDecimal INTERVAL = BigDecimal.valueOf(0.2);
-  private static final BigDecimal UPPER_END = BigDecimal.valueOf(UPPER_END_VAL);
-  private static final BigDecimal LOWER_END = BigDecimal.valueOf(LOWER_END_VAL);
+  private static final double LOWER_SCORE_VAL = -1.0;
+  private static final double UPPER_SCORE_VAL = 1.0;
+  private static final double SCORE_INTERVAL_VAL = 0.2;
+  private static final BigDecimal SCORE_INTERVAL = BigDecimal.valueOf(SCORE_INTERVAL_VAL);
+  private static final BigDecimal UPPER_SCORE = BigDecimal.valueOf(UPPER_SCORE_VAL);
+  private static final BigDecimal LOWER_SCORE = BigDecimal.valueOf(LOWER_SCORE_VAL);
+
   private static final float TEST_SCORE = 0.23f;
+  private static final float TEST_MAGNITUDE = 1.5f;
 
   /**
-   * It constructs a HashMap with current range for expected score categorizations for comparisions.
+   * It constructs a List with sentiment bucket for expected score categorizations, frequency and
+   * top comments for testing
    *
+   * @param userCommentList a list of userComment corresponding to each interval that have top N
+   *     magnitude
    * @param frequency a list of frequency corresponding to each interval sorted in ascending order
    * @return constructed hashmap with keys as ranges based on intervals and values corresponding to
    *     frequency input
    */
-  private Map<Range, Integer> constructRangeMapFromFrequencyList(List<Integer> frequency) {
-    if (frequency.size()
-        != UPPER_END.subtract(LOWER_END).divide(INTERVAL, RoundingMode.UP).intValue()) {
+  private List<SentimentBucket> constructRangeMapFromFrequencyList(
+      List<List<UserComment>> userCommentList,
+      List<Integer> frequency,
+      BigDecimal lowerEnd,
+      BigDecimal upperEnd,
```

Comment:
I think we could get away with removing the `frequency` parameter since the length of each individual list w/in the `userCommentList` should serve the same purpose.

In doing so, maybe we should rename the method to `constructSentimentBucketListFromBucketedCommentList` (or something not as horribly long or convoluted).

Reply:
I don't thinkg the length of list of userComent is equal to the number from frequency list....

Reply:
We can talk about this in our 1-1 later today~